torch/_dynamo/variables/builder.py

```diff
@@ -1498,6 +1500,135 @@ def wrap_symint(self, value):
 
         return unspec_var
 
+    def wrap_symfloat(self, value):
+        # SymFloat wrapping is special.  We first wrap it in the same way we
+        # do an unspecialized primitive, and then we item() it into a
+        # SymFloat.  Removal of the item() call is left to a later FX pass,
+        # mostly because that pass is more easily done after decomposition
```

Comment:
Hmm, what's not clear? The pass is eventually going to go in https://github.com/pytorch/pytorch/pull/125423 but I haven't entirely figured out all the details of how it will work. In broad strokes though, if we have

```
def f(y: f32[10], x: f64[]):
  zf0 = x.item()
  zf1 = zf0 + 2
  z = y + zf1
  return z
```

I want to "push" the item() down so it gets absorbed into the addition

```
def f(y: f32[10], x: f64[]):
  x1 = x + 2
  z = y + x1
  return z
```

now no more item() at all.

Reply:
Ah. The part I didn't understand is the "that pass is more easily done after decomposition". I guess you mean after lowering to ATen ops? Not fully sure what you mean by "decomposition"

Reply:
Yeah, lowering to ATen ops. Although, I'm flip flopping on this again.

One compelling reason to do the pass directly on the Dynamo graph, or perhaps even as Dynamo is executing, is situations where you have an op overload `f.float(Tensor, float)` and `f.Tensor(Tensor, Tensor)`. The translation must happen BEFORE we do fake tensor propagation over this function at all, because if you fake tensor propagate over f.float, you will immediately force specialization on the float argument (because we don't support float arguments).

But maybe this is a bit hypothetical. In particular, Scalar overloads don't have this problem (since they can directly accept SymFloat).  Or maybe we ought to just unconditionally support SymFloat input to float and try not to specialize it unless we have to turn it into a C++ double.

Reply:
Note that, at the moment, we don't have any test that makes sure that those two overloads are equivalent, so this is bound to uncover an interesting amount of bugs.

Leaving that aside, I think the right way forward is to support SymFloat overloads, the same way we symintified eager.

Reply:
Well, you can't do the conversion automatically. I have to do it manually one by one. But OK, I guess we'll solve specs this way.